File: src/main/java/com/efedotov/vaultly/dto/folder/FolderShareDto.java
package com.efedotov.vaultly.dto.folder;

import lombok.Data;

import java.util.UUID;
import java.util.List;

@Data
public class FolderShareDto {
    private UUID folderId;
    private List<UUID> userIds;
    private Boolean canEdit = false;
    private Boolean canDelete = false;
    private Boolean canShare = false;
    private String message;
}

File: lib/server/model/folder/folder_share_dto/folder_share_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'folder_share_dto.freezed.dart';
part 'folder_share_dto.g.dart';

@freezed
abstract class FolderShareDto with _$FolderShareDto {
  const factory FolderShareDto({
    String? folderId,
    List<String>? userIds,
    bool? canEdit,
    bool? canDelete,
    bool? canShare,
    String? message,
  }) = _FolderShareDto;

  factory FolderShareDto.fromJson(Map<String, dynamic> json) =>
      _$FolderShareDtoFromJson(json);
}